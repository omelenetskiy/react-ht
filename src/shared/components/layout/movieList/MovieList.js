import React from 'react';
import styled from 'styled-components';

import MovieCard from '../movieCard/MovieCard';

const StyledMovieCardWrapper = styled.div`
  display: flex;
  padding: 15px 0;
  flex-flow: row wrap;
  justify-content: center;
  background: #fff;
`;

const MovieList = ({ movies }) => (
  <StyledMovieCardWrapper>
    {movies && movies.length ? (
      movies.map(movie => {
        const date = Date.parse(movie.release_date);
        const parsedDate = new Date(date).getFullYear();
        return (
          <MovieCard key={movie.id} movie={movie} parsedDate={parsedDate} />
        );
      })
    ) : (
      <p>No films found</p>
    )}
  </StyledMovieCardWrapper>
);

export default MovieList;
