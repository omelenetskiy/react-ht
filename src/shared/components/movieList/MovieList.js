import React from 'react';
import styled from 'styled-components';
import { connect } from 'react-redux';
import { getMoviesState } from '../../store/selectors';

import MovieCard from '../movieCard/MovieCard';

const StyledMovieCardWrapper = styled.div`
  display: flex;
  padding: 15px 0;
  flex-flow: row wrap;
  justify-content: center;
  align-items: center;
  background: rgba(255, 255, 255, 0.7);
  flex-grow: 0.9;

  .not-found {
    font-size: 45px;
    font-weight: bold;
  }
`;

type Props = {
  movies: Array<{
    release_date: string,
    id: string,
  }>,
};

const MovieList = ({ movies }: Props) => (
  <StyledMovieCardWrapper>
    {movies.length ? (
      movies.map((movie) => {
        const date = Date.parse(movie.release_date);
        const parsedDate = new Date(date).getFullYear();
        return (
          <MovieCard key={movie.id} movie={movie} parsedDate={parsedDate} />
        );
      })
    ) : (
      <p className="not-found">No movies found :(</p>
    )}
  </StyledMovieCardWrapper>
);

const mapStateToProps = (state) => ({
  movies: getMoviesState(state),
});

export default connect(mapStateToProps)(MovieList);
