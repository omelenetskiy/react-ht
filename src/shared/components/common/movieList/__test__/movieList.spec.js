import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import "jest-enzyme";
import "jest-styled-components";

import MovieList from "../MovieList";

const props = {
  parsedDate: "parsedDate",
  movies: [
    {
      id: "id",
      title: "title",
      poster_path: "poster_path",
      release_date: "release_date",
      genres: ["genre-1", "genre-2"]
    }
  ]
};

describe("MovieList", () => {
  it("should render correctly", () => {
    const wrapper = (
      <Router>
        <MovieList {...props} />
      </Router>
    );

    const tree = renderer.create(wrapper).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
