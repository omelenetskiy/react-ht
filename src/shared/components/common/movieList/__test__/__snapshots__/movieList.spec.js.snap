// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList should render correctly 1`] = `
.c2 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #fff;
}

.c2 .card-img {
  width: 100%;
  height: 300px;
}

.c2 .card-body {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 5px;
}

.c2 .card-body .card-title {
  margin: 0;
  font-size: 14px;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  text-transform: uppercase;
}

.c2 .card-body .badge {
  font-size: 12px;
  border: 1px solid;
  border-radius: 2px;
  padding: 2px;
}

.c2 .card-footer {
  max-width: 100%;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  font-size: 12px;
  padding: 5px;
}

.c1 {
  border: 1px solid;
  background-color: #fff;
  border-radius: 3px;
  margin: 5px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 15px 0;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: rgba(255,255,255,0.7);
}

<div
  className="c0"
>
  <a
    className="c1"
    href="/movie/id"
    onClick={[Function]}
  >
    <div
      className="c2"
    >
      <img
        alt="title"
        className="card-img"
        src="poster_path"
      />
      <div
        className="card-body"
      >
        <h5
          className="card-title"
        >
          title
        </h5>
        <span
          className="badge"
        >
          NaN
        </span>
      </div>
      <div
        className="card-footer"
      >
        genre-1 & genre-2
      </div>
    </div>
  </a>
</div>
`;
